### ISSUE TITLE
[Enhancement] Info Panel: Only auto-popup when an F#-type file is opened.

### ISSUE THREAD

**Issue by baronfel** [ionide/ionide-vscode-fsharp#1176]
**Is your feature request related to a problem? Please describe.**
When I open my work repository and I was last working on yaml/bash/any non-F# file, the Info Panel pops up.  When I'm working with F# files I want this behavior, but for non-F# coding tasks I don't want it right away.

**Describe the solution you'd like**
I'd like if the Info Panel wasn't popped-up automatically until the first F# file was opened.  This would keep the Info Panel relevant while minimizing the number of times it's present and not immediately useful, I think.  Imagine the case where you open a yaml file first: the info panel is present and completely blank, obscuring parts of your screen. Now imagine a world where you work in yaml unobstructed until you go to some F# code and BAM your beloved Info Panel is right there with you, helping you along your coding journey.

**Describe alternatives you've considered**
Closing the Info Panel all the time. This hurts because it's like I'm saying goodbye to a close friend.

**Comment by Krzysztof-Cieslak** [comment]
Makes sense, should be fairly easy to do - in the place where we make InfoPanel visible during plugin startup we can check current list of opened documents and check if any is F# file.

**Comment by teo-tsirpanis** [comment]
Another option is to never auto-popup, or even better, to configure its behavior through the settings.

**Comment by MangelMaxime** [comment]
> Another option is to never auto-popup, or even better, to configure its behavior through the settings.

You can already configure it's behaviour via the settings:

<img width="588" alt="Capture d’écran 2019-07-15 à 12 18 34" src="https://user-images.githubusercontent.com/4760796/61209683-b3cb7100-a6fa-11e9-8f88-6da143acd26e.png">

**Comment by teo-tsirpanis** [comment]
That’s exactly how my setting look like. Doesn’t the second option being disabled imply that the F# window shouldn’t popup on start?

Από: Maxime Mangel
Αποστολή: Δευτέρα, 15 Ιουλίου 2019 1:19 μμ
Προς: ionide/ionide-vscode-fsharp
Κοιν.: Theodore Tsirpanis; Comment
Θέμα: Re: [ionide/ionide-vscode-fsharp] [Enhancement] Info Panel: Onlyauto-popup when an F#-type file is opened. (#1176)

Another option is to never auto-popup, or even better, to configure its behavior through the settings.
You can already configure it's behaviour via the settings:

**Comment by MangelMaxime** [comment]
You are confusing the F# solution explorer (on the left) and the Info Panel (on the right).

<img width="1920" alt="Capture d’écran 2019-07-15 à 12 45 12" src="https://user-images.githubusercontent.com/4760796/61210996-94364780-a6fe-11e9-9a07-1af73887c31b.png">

If you don't want to show "F# Solution explorer"  on start up you can use:

<img width="701" alt="Capture d’écran 2019-07-15 à 12 49 17" src="https://user-images.githubusercontent.com/4760796/61211136-027b0a00-a6ff-11e9-864c-6c8aaff9b7c2.png">

**Comment by teo-tsirpanis** [comment]
OK. Thanks.

**Comment by ZaymonFC** [comment]
Is this issue still open?

**Comment by wilsoncg** [comment]
I too have noticed this. I open up a csharp project, and within a few seconds the F# solution explorer & InfoPanel appear all proud as punch, only to steal my screen real estate and then tell me csproj is not a supported language. 🙄

If you close the info panel and right click -> hide the F# solution explorer, close & re-open vscode **csproj** solution, this makes no difference as they both popup again.

![fsharp-sln-explorer-look-at-me](https://user-images.githubusercontent.com/1066103/99587635-4069fb80-29e1-11eb-8d1e-d59a665c8254.gif)

Looking at the vscode docs I notice this key piece of information under [Extension Host](https://code.visualstudio.com/api/advanced-topics/extension-host):

> For example, the Markdown extension should only be loaded when a user opens a Markdown file. This makes sure that extensions do not consume unnecessary CPU and memory.

Under [Activation Events](https://code.visualstudio.com/api/advanced-topics/extension-host):

> This activation event is emitted and interested extensions will be activated whenever a file that resolves to a certain language gets opened.
The `onLanguage` event takes a [language identifier](https://code.visualstudio.com/docs/languages/identifiers) value.

@Krzysztof-Cieslak would using the onLanguage activation event be a better approach for initialising the info panel & solution explorer? Could also do away with the info panel on startup setting as it would then be redudant.

I'd be happy to try submitting a PR if you can sort of point me in the right direction, is it something in the `activate()` function in [InfoPanel.fs](https://github.com/ionide/ionide-vscode-fsharp/blob/master/src/Components/InfoPanel.fs)?

**Comment by baronfel** [comment]
For reference, @wilsoncg, the current set of activation events that we use are located [here](https://github.com/ionide/ionide-vscode-fsharp/blob/master/release/package.json#L1652-L1659). We're already using the fsharp language identifier, which is not the problem here. We don't want to change these because we want Ionide to be running before you need it and these do a good job for this.

I believe the fix as @Krzysztof-Cieslak was suggesting is to expand the condition [here](https://github.com/ionide/ionide-vscode-fsharp/blob/master/src/Components/InfoPanel.fs#L328) to check if [`window.visibleTextEditors`](https://code.visualstudio.com/api/references/vscode-api#window) contains any F#-related files before opening. So that condition would become something like:

```fsharp
if show && window.visibleTextEditors |> Seq.exists isFsharpTextEditor 
then openPanel () |> ignore
else ()
```

**Comment by wilsoncg** [comment]
Interesting @baronfel. I'm still slightly confused, as vscode has correctly detected my project is C# but then ionide has ignored that & forcefully loaded anyway. That looks to me like ionide is not correctly using the language identifier activation.

For example as a contrast, the docker extension looks to load on demand. The F# solution explorer is loading for C# projects, & grabbing focus, should it not load on demand instead? Surely it's un-necessary to do any loading for a C# project?

**Comment by baronfel** [comment]
The part I think you're overlooking is that solution files are part of our activation events as well. There are many different activation events described in the links I sent you, and it's this one that is causing the overlap on c# projects.  You'll note that omnisharp for example has [the same activation event](https://github.com/OmniSharp/omnisharp-vscode/blob/master/package.json#L427), loading whenever an F# only solutions load, so you have the same problem.

**Comment by wilsoncg** [comment]
Ah right, got it, thanks for the info.

**Comment by baronfel** [comment]
personally, I have things set up to fold the solution explorer into the outline pane instead of a separate pane, which seems to do a good job of preventing that pop-out you speak of.  I think that coupled with the code change above on the info panel would go a long way to handling your concerns.

**Comment by Krzysztof-Cieslak** [comment]
Fixed by #1438
